internal/docstore/docstore.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package docstore provides a portable implementation of a document store.
```

Comment:
(fine to add a TODO for this)

Maybe add reference to explain what a "document store" is (https://en.wikipedia.org/wiki/Document-oriented_database ?) Or explain it here.

Also, please update the package docstring to look more like `blob`'s -- e.g., Errors section, explaining about concrete type, etc. It's copypasta but I think useful for new users.

Reply:
Added TODOs.